gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,246 @@
+        return this._sliderThumbMaterial;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new TransformNode(`${this.name}_slider`, scene);
+
+        const sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);
+        sliderBar.rotation.z = -Math.PI / 2;
+        sliderBar.scaling.y = SLIDER_SCALING;
+        sliderBar.isPickable = false;
+        sliderBar.setParent(anchor);
+
+        const sliderThumb = MeshBuilder.CreateBox(`${this.name}_sliderthumb`, { size: 0.1 }, scene);
+        sliderThumb.scaling = new Vector3(SLIDER_SCALING, SLIDER_SCALING, SLIDER_SCALING);
+        sliderThumb.position.x = this._convertToPosition(this.value);
+        sliderThumb.addBehavior(this._createBehavior());
+        sliderThumb.setParent(anchor);
+
+        this._sliderBar = sliderBar;
+        this._sliderThumb = sliderThumb;
+
+        return anchor;
+    }
+
+    protected _affectMaterial(mesh: AbstractMesh) {
+        const barMaterial = new StandardMaterial(`${this.name}_sliderbar_material`, mesh.getScene());
+        barMaterial.specularColor = Color3.Black();
+        this._sliderBar.material = barMaterial;
+
+        const thumbMaterial = new StandardMaterial(`${this.name}_sliderthumb_material`, mesh.getScene());
+        thumbMaterial.specularColor = Color3.Black();
+        mesh.material = thumbMaterial;
+
+        this._sliderBarMaterial = barMaterial;
+        this._sliderThumbMaterial = thumbMaterial;
+    }
+
+    private _createBehavior(): PointerDragBehavior {
+        const pointerDragBehavior = new PointerDragBehavior({ dragAxis: new Vector3(1, 0, 0) });
+        pointerDragBehavior.moveAttached = false;
+
+        pointerDragBehavior.onDragObservable.add((event) => {
+            const newPosition = this._sliderThumb.position.x + event.dragDistance;
+            this._sliderThumb.position.x = Math.max(Math.min(newPosition, this.end), this.start);
+            this.value = this._convertToValue(this._sliderThumb.position.x);
+        });
+
+        pointerDragBehavior.onDragEndObservable.add((event) => {
+            this._sliderThumb.position.x = this._convertToPosition(this.value);
+        });
+
+        return pointerDragBehavior;
+    }
+
+    private _convertToPosition(value: number): number {
+        const position = ((value - this.minimum) / (this.maximum - this.minimum)) * (this.end - this.start) + this.start;
+        return Math.min(Math.max(position, this.start), this.end);
+    }
+
+    private _convertToValue(position: number): number {
+        let value = ((position - this.start) / (this.end - this.start)) * (this.maximum - this.minimum) + this.minimum;
+        value = this.step ? Math.round(value / this.step) * this.step : value;
+
+        return Math.max(Math.min(this.roundingToDecimal ? Math.round(value * 100) / 100 : value, this._maximum), this._minimum);

```

Comment:
Why is the decimal rounding done to two decimal places?

Reply:
No specific reason. I looked up the MRTK slider example, which uses two decimal places for values. One idea I have is to let the use set this as an option. We can default it to two decimals and we don't even have to check if  we need rounding or not (I am using the min, max or step values to determine that)